lighthouse-core/gather/gatherers/start-url.js

```diff
@@ -75,19 +81,25 @@ class StartUrl extends Gatherer {
     // Wait up to 3s to get a matched network request from the fetch() to work
     const timeoutPromise = new Promise(resolve =>
       setTimeout(
-        () => resolve({statusCode: -1, explanation: 'Timed out waiting for start_url to respond.'}),
+        () => resolve({
+          statusCode: -1,
+          explanation: `Timed out waiting for start_url (${startUrl}) to respond.`,
+        }),
         3000
       )
     );
 
+    const startUrlObject = new URL(startUrl);
+    const startUrlNoHash = startUrlObject.origin + startUrlObject.pathname;
```

Comment:
reconstructing URLs with concatenation is super frightening to me.

```suggestion
    startUrlObject.hash = '';
    const startUrlNoHash = startUrlObject.toString();
```

Reply:
ya this smarrtyer